applications/nrf_desktop/src/modules/dfu_mcumgr.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/dfu/mcuboot.h>
+#include <zephyr/mgmt/mcumgr/mgmt/callbacks.h>
+#include <zephyr/sys/math_extras.h>
+
+#include <app_event_manager.h>
+#include "dfu_lock.h"
+
+#define MODULE dfu_mcumgr
+#include <caf/events/module_state_event.h>
+#include <caf/events/ble_smp_event.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_DFU_MCUMGR_LOG_LEVEL);
+
+#define DFU_TIMEOUT	K_SECONDS(5)
+
+static struct k_work_delayable dfu_timeout;
+
+static atomic_t mcumgr_event_active = ATOMIC_INIT(false);
+
+/* Declare the function here as it is not available in the MCUmgr API. */
+void img_mgmt_reset_upload(void);
+
+static void dfu_lock_owner_changed(const struct dfu_lock_owner *new_owner)
+{
+	LOG_DBG("MCUmgr progress reset due to different DFU process operations");
+
+	img_mgmt_reset_upload();
+}
+
+const static struct dfu_lock_owner mcumgr_owner = {
+	.name = "MCUmgr",
+	.owner_changed = dfu_lock_owner_changed,
+};
+
+static void dfu_timeout_handler(struct k_work *work)
+{
+	LOG_WRN("MCUmgr DFU timed out");
+
+	if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK)) {
+		int err;
+
+		err = dfu_lock_release(&mcumgr_owner);
+		__ASSERT_NO_MSG(!err);
+	}
+}
+
+static int32_t mcumgr_img_mgmt_cb(uint32_t event,
+				  int32_t rc,
+				  bool *abort_more,
+				  void *data,
+				  size_t data_size)
+{
+	LOG_INF("MCUmgr Image Management Event with the %d ID",
+		u32_count_trailing_zeros(MGMT_EVT_GET_ID(event)));
+
+	k_work_reschedule(&dfu_timeout, DFU_TIMEOUT);
+	if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK) && dfu_lock_claim(&mcumgr_owner)) {
+		(void)k_work_cancel_delayable(&dfu_timeout);
+		return MGMT_ERR_EACCESSDENIED;
+	}
+
+	if (atomic_cas(&mcumgr_event_active, false, true)) {
+		APP_EVENT_SUBMIT(new_ble_smp_transfer_event());
+	}
```

Comment:
Maybe the event could be submitted only if the mcumgr uses BLE transport? Your implementation does not enforce transport (unlinke CAF's implementation)